src/ssh/cockpitsshrelay.c

```diff
@@ -477,35 +436,29 @@ write_tmp_knownhosts_file (CockpitSshData *data,
                            const gchar *content,
                            const gchar **problem)
 {
-  FILE *fp = NULL;
+  int fd;
+  g_autoptr(GError) error = NULL;
 
-  tmp_knownhost_file = create_knownhosts_temp ();
-  if (!tmp_knownhost_file)
-    {
-      *problem = "internal-error";
-      return FALSE;
-    }
-  atexit (cleanup_knownhosts_file);
-
-  fp = fopen (tmp_knownhost_file, "a");
-  if (fp == NULL)
+  fd = g_file_open_tmp ("known-hosts.XXXXXX", &tmp_knownhost_file, &error);
+  if (fd < 0)
     {
       g_warning ("%s: couldn't open temporary known host file for data: %s",
-                 data->logname, tmp_knownhost_file);
+                 data->logname, error->message);
       *problem = "internal-error";
       return FALSE;
     }
+  close (fd);
```

Comment:
Maybe add a comment here about now that the `O_EXCL`-created file exists with the given name, the security-sensitive work of `g_file_open_tmp()` is done, and we're free to use the name below?

Reply:
Added.